File: leetcode/1630. Arithmetic Subarrays/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/arithmetic-subarrays/?envType=study-plan&id=programming-skills-ii


class Solution {
  public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
    List<Boolean> ans = new ArrayList<>();
    int[] subarr;
    int n = l.length;

    for (int i = 0; i < n; i++) {
      subarr = Arrays.copyOfRange(nums, l[i], r[i] + 1);
      ans.add(checkArithmetic(subarr));
    }

    return ans;
  }

  private boolean checkArithmetic(int[] arr) {
    if (arr.length <= 2) return true;

    Arrays.sort(arr);
    int d = arr[1] - arr[0];
    System.out.println(d);

    for (int i = 2; i < arr.length; i++) {
      if (arr[i] - arr[i-1] != d) {
        return false;
      }
    }

    return true;
  }
}

File: leetcode/1630. Arithmetic Subarrays/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/arithmetic-subarrays/

const checkArithmetic = (arr: number[]): boolean => {
  arr.sort((a, b) => a - b)
  let d = arr[1] - arr[0]
  for (let i = 2; i < arr.length; i++) {
    if (arr[i] - arr[i - 1] != d) {
      return false
    }
  }
  return true
}

const checkArithmeticSubarrays = (
  nums: number[],
  l: number[],
  r: number[]
): boolean[] => {
  let ans: boolean[] = [],
    flag = true
  for (let i = 0; i < l.length; i++) {
    flag = checkArithmetic(nums.slice(l[i], r[i] + 1))
    ans.push(flag)
  }
  return ans
}

console.log(checkArithmeticSubarrays([4, 6, 5, 9, 3, 7], [0, 0, 2], [2, 3, 5]))
